test/com/tfl/billing/JourneyCostCalculatorTest.java

```diff
@@ -2,70 +2,120 @@
 
 import com.tfl.external.Customer;
 import com.tfl.external.CustomerDatabase;
+import com.tfl.underground.OysterReaderLocator;
+import com.tfl.underground.Station;
+import org.jmock.Expectations;
 import org.jmock.integration.junit4.JUnitRuleMockery;
+import org.junit.Before;
 import org.junit.BeforeClass;
 import org.junit.Rule;
 import org.junit.Test;
 
 import java.math.BigDecimal;
-import java.util.Date;
-import java.util.List;
+import java.util.*;
 
 import static org.hamcrest.CoreMatchers.is;
-import static org.junit.Assert.assertFalse;
-import static org.junit.Assert.assertThat;
-import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.*;
 
 
 public class JourneyCostCalculatorTest {
-    JourneyCostCalculator testCalc = new JourneyCostCalculator();
+
+
+    @Rule
+    public JUnitRuleMockery context = new JUnitRuleMockery();
+
+    //testing assets
     Date peakTime = new Date();
     Date offPeakTime = new Date();
     Customer c = CustomerDatabase.getInstance().getCustomers().get(0);
+    Customer c1 = CustomerDatabase.getInstance().getCustomers().get(1);
+    Customer c2 = CustomerDatabase.getInstance().getCustomers().get(2);
+
+    //initilaize required object
+    private List<JourneyEvent> eventLog = Arrays.asList(new JourneyEvent(c.cardId(), OysterReaderLocator.atStation(Station.PADDINGTON).id()), new JourneyEvent(c1.cardId(), OysterReaderLocator.atStation(Station.PADDINGTON).id()), new JourneyEvent(c.cardId(), OysterReaderLocator.atStation(Station.VICTORIA_STATION).id()), new JourneyEvent(c.cardId(), OysterReaderLocator.atStation(Station.OXFORD_CIRCUS).id()));
+    private List<Journey> customerJourneys = Arrays.asList(new Journey(new JourneyEvent(c.cardId(), OysterReaderLocator.atStation(Station.PADDINGTON).id()), new JourneyEvent(c.cardId(), OysterReaderLocator.atStation(Station.VICTORIA_STATION).id())));
+    JourneyCostCalculator costCalcuator = new JourneyCostCalculator(eventLog, customerJourneys);
+
 
-    @Rule
-    public JUnitRuleMockery context = new JUnitRuleMockery();
 
     private long getTimeInMillis(int hour) {
         return hour*60*60*1000;
     }
 
     @Test
-    public void getJourneyEventsTest() throws Exception {
-        List<JourneyEvent> events = context.mock(List.class);
-
+    public void getAllCardScansForCustomer() throws Exception {
```

Comment:
why would this method throw exeptions? Maybe consider making another test which ensures that no exception is thrown? 
https://stackoverflow.com/questions/17731234/how-to-test-that-no-exception-is-thrown